src/rhsm/connection.py

```diff
@@ -131,6 +133,24 @@ def __str__(self):
         return "Bad certificate at %s" % self.cert_path
 
 
+class ConnectionType(enum.Flag):
```

Comment:
considering the various connection types are mutually exclusive, IMHO the type should be `Enum`; in addition, if the values will be the description/debug strings, it is possible to use the automatic string representation to avoid the manual string assignment on debug (see later on)

Reply:
I agree that it should be `enum.Enum` and not `enum.Flag`. I don't to assign some values to `BASIC_AUTH`, `NO_AUTH`, etc. that are used for debug print in one method in another class.

Reply:
> I don't to assign some values to `BASIC_AUTH`, `NO_AUTH`, etc. that are used for debug print in one method in another class.

I don't understand this sentence, sorry :-/

What I mean is that you can use strings as values for the enum items, and use them when printing for debug:
```python
>>> class FooEnum(enum.Enum):
...    FOO_1 = "foo_1"
...    FOO_2 = "foo2?"
>>> x = FooEnum.FOO_1
>>> print(x.value)
foo_1
```
This way, in the (remote) case of adding a new authentication type, it is only needed to add the new enum item without adding it also to debug blocks.

Reply:
Yeah, your are proposing and I'm not accepting. I opened this PR almost 2 months ago. Is it something critical that is blocking this PR from merging to main?